api/v0/ingest/schema/schema.go

```diff
@@ -0,0 +1,83 @@
+package schema
+
+import (
+	_ "embed"
+	"fmt"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime"
+	_ "github.com/ipld/go-ipld-prime/codec/dagjson"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+	"github.com/ipld/go-ipld-prime/schema"
+	"github.com/multiformats/go-multicodec"
+	"github.com/multiformats/go-multihash"
+)
+
+const (
+	// MaxContextIDLen specifies the maximum number of bytes accepted as Advertisement.ContextID.
+	MaxContextIDLen = 64
+	// MaxMetadataLen specifies the maximum number of bytes an advertisement metadata can contain.
+	MaxMetadataLen = 1024 // 1KiB
+)
+
+var (
+	// NoEntries is a special value used to explicitly indicate that an
+	// advertisement does not have any entries. When isRm is true it and serves to
+	// remove content by context ID, and when isRm is false it serves to update
+	// metadata only.
+	NoEntries cidlink.Link
+
+	// Linkproto is the ipld.LinkProtocol used for the ingestion protocol.
+	// Refer to it if you have encoding questions.
+	Linkproto = cidlink.LinkPrototype{
+		Prefix: cid.Prefix{
+			Version:  1,
+			Codec:    uint64(multicodec.DagJson),
+			MhType:   uint64(multicodec.Sha2_256),
+			MhLength: 16,
+		},
+	}
+
+	// TypeAdvertisement represents the IPLD schema type for an advertisement.
+	//
+	// See: Advertisement.
+	TypeAdvertisement schema.Type
+
+	// AdvertisementPrototype represents the IPLD node prototype of Advertisement.
+	//
+	// See: bindnode.Prototype.
+	AdvertisementPrototype ipld.NodePrototype
```

Comment:
FYI these could be schema.TypedPrototype, which also lets you e.g. decode at the representation level. I actually imagine that's what you want, as otherwise you're decoding at the type level, ignoring e.g. representation strategies.